### PR TITLE
feat(project): support drag and drop supported file in room

### PR DESCRIPTION
**Pull request by hyrious**

### CODE REVIEW

File: web/flat-web/src/utils/dnd/index.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { getFileExt } from "../file";
```
Comment (BlackHole1) [inline web/flat-web/src/utils/dnd/index.ts]: what is dnd?

File: web/flat-web/src/utils/dnd/index.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+const SupportedFileExts = new Set(
```
Comment (BlackHole1) [inline web/flat-web/src/utils/dnd/index.ts]: why not use `array`?

**Review by BlackHole1 — CHANGES_REQUESTED** [review]
miss desktop?

File: web/flat-web/src/utils/dnd/index.ts (hunk 1 above)
Comment (hyrious) [reply web/flat-web/src/utils/dnd/index.ts]: Drag aNd Drop

File: web/flat-web/src/utils/dnd/index.ts (hunk 2 above)
Comment (hyrious) [reply web/flat-web/src/utils/dnd/index.ts]: set is faster..

Comment (BlackHole1) [reply web/flat-web/src/utils/dnd/index.ts]: But you used split

File: web/flat-web/src/utils/dnd/index.ts (hunk 1 above)
Comment (BlackHole1) [reply web/flat-web/src/utils/dnd/index.ts]: replace: `drag-and-drop`

File: desktop/renderer-app/src/pages/OneToOnePage/index.tsx [hunk 3]
```diff
@@ -72,7 +72,8 @@ export const OneToOnePage = observer<OneToOnePageProps>(function OneToOnePage()
 
     const params = useParams<RouteParams<RouteNameType.OneToOnePage>>();
 
-    const classRoomStore = useClassRoomStore(params.roomUUID, params.ownerUUID, recordingConfig);
+    const { i18n, t } = useTranslation();
+    const classRoomStore = useClassRoomStore({ ...params, recordingConfig, i18n });
```
Comment (BlackHole1) [inline desktop/renderer-app/src/pages/OneToOnePage/index.tsx]: wrap line

File: desktop/renderer-app/src/stores/class-room-store.ts [hunk 4]
```diff
@@ -1033,15 +1050,23 @@ export class ClassRoomStore {
...
+export function useClassRoomStore({
+    roomUUID,
+    ownerUUID,
+    recordingConfig,
+    classMode,
+    i18n,
+}: ClassRoomStoreConfig): ClassRoomStore {
     const [classRoomStore] = useState(
-        () => new ClassRoomStore({ roomUUID, ownerUUID, recordingConfig, classMode }),
+        () => new ClassRoomStore({ roomUUID, ownerUUID, recordingConfig, classMode, i18n }),
```
Comment (BlackHole1) [inline desktop/renderer-app/src/stores/class-room-store.ts]: wrap line

File: desktop/renderer-app/src/stores/whiteboard-store.ts [hunk 5]
```diff
@@ -430,4 +452,211 @@ export class WhiteboardStore {
...
         return scenePath.slice(0, scenePath.lastIndexOf("/"));
     };
+
+    public insertCourseware = async (file: CloudStorageFile): Promise<void> => {
+        if (file.convert === "converting") {
+            void message.warn(this.i18n.t("in-the-process-of-transcoding-tips"));
+            return;
+        }
+
+        void message.info(this.i18n.t("Inserting-courseware-tips"));
```
Comment (BlackHole1) [inline desktop/renderer-app/src/stores/whiteboard-store.ts]: to lower case

File: desktop/renderer-app/src/utils/drag-and-drop/index.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+const SupportedFileExts =
+    ".ppt,.pptx,.doc,.docx,.pdf,.png,.jpg,.jpeg,.gif,.mp3,.mp4,.ice,.vf".split(",");
```
Comment (BlackHole1) [inline desktop/renderer-app/src/utils/drag-and-drop/index.ts]: to ["ppt", .....];

**Review by BlackHole1 — CHANGES_REQUESTED** [review]
wrap line

**Review by BlackHole1 — APPROVED** [review]
nice！

**Comment by BlackHole1** [comment]
please change commit and pr title scope to: project

**Comment by hyrious** [comment]
You can squash and change the commit message.

**Pull request merged**